just_audio/example/lib/example_playlist.dart

```diff
@@ -328,26 +328,27 @@ class ControlButtons extends StatelessWidget {
             final playerState = snapshot.data;
             final processingState = playerState?.processingState;
             final playing = playerState?.playing;
-            if (processingState == ProcessingState.loading ||
-                processingState == ProcessingState.buffering) {
-              return Container(
-                margin: EdgeInsets.all(8.0),
-                width: 64.0,
-                height: 64.0,
-                child: CircularProgressIndicator(),
-              );
-            } else if (playing != true) {
+            if (playing == false || playing == null) {
               return IconButton(
                 icon: Icon(Icons.play_arrow),
                 iconSize: 64.0,
                 onPressed: player.play,
               );
-            } else if (processingState != ProcessingState.completed) {
+              // } else if (processingState != ProcessingState.completed) {
+            } else if (playing == true) {
               return IconButton(
                 icon: Icon(Icons.pause),
                 iconSize: 64.0,
                 onPressed: player.pause,
               );
+            } else if (processingState == ProcessingState.loading ||
+                processingState == ProcessingState.buffering) {
+              return Container(
+                margin: EdgeInsets.all(8.0),
+                width: 64.0,
+                height: 64.0,
+                child: CircularProgressIndicator(),
+              );
```

Comment:
I moved this down here because the indicator was spinning forever.

Reply:
I would suggest leaving this example unchanged as a sort of litmus test for the correct behaviour. But perhaps you could include an `example` subdirectory within the just_audio_libwinmedia repo. Regarding the bug itself, this would indicate that `processingState` never reaches `ready`.

Reply:
Indeed I never called the ready enum!